R/ml_clustering_kmeans.R

```diff
@@ -133,18 +133,23 @@ new_ml_kmeans <- function(jobj) {
 
 new_ml_kmeans_model <- function(jobj) {
   summary <- possibly_null(~ new_ml_kmeans_summary(invoke(jobj, "summary")))()
-  new_ml_clustering_model(
+  kmeans_model <- new_ml_clustering_model(
     jobj,
     # `def clusterCenters`
     cluster_centers = possibly_null(
       ~ invoke(jobj, "clusterCenters") %>%
         purrr::map(invoke, "toArray")
     ),
-    compute_cost = function(dataset) {
-      invoke(jobj, "computeCost", spark_dataframe(dataset))
-    },
     summary = summary,
     class = "ml_kmeans_model")
+
+  if (!is_required_spark(jobj, "3.0.0")) {
```

Comment:
I think we would want to warn here for Spark >= 3.0.0, something like:

```
warning("The 'compute_cost' parameter is no longer supported in Spark 3.x")
```

If we think there might be a few more parameter deprecations, worth calling and adding:

```
spark_param_deprecated <- function(param, version = "3.x") {
  warning("The '", param, "' parameter is deprecated in Spark ", version)
}
```